iceberg_rust_ffi/src/catalog.rs

```diff
@@ -129,11 +135,11 @@ impl IcebergCatalog {
     pub fn set_token_authenticator(
         &mut self,
         callback: CustomAuthenticatorCallback,
```

Comment:
Should we construct FFITokenAuthenticator outside of `set_token_authenticator` and then just set it here?